catalog/core/catalog-core-dynamic/src/main/java/ddf/catalog/data/dynamic/impl/DynamicMetacard.java

```diff
@@ -0,0 +1,644 @@
+     */
+    @Override
+    public String getTitle() {
+        return getPropertyAsString(Metacard.TITLE);
+    }
+
+    /**
+     * Returns the resource URI of this {@link Metacard}. <br/>
+     * Convenience method for <code>
+     * {@link #getAttribute}({@link Metacard#RESOURCE_URI})
+     * </code>
+     *
+     * @return the resource URI for this {@link Metacard}
+     */
+    @Override
+    public URI getResourceURI() {
+        return getPropertyAsURI(Metacard.RESOURCE_URI);
+    }
+
+    /**
+     * Returns the resource size for this {@link Metacard}. <br/>
+     * Convenience method for <code>
+     * {@link #getAttribute}({@link Metacard#RESOURCE_URI})
+     * </code>
+     *
+     * @return the resource size for this {@link Metacard}
+     */
+    @Override
+    public String getResourceSize() {
+        return getPropertyAsString(Metacard.RESOURCE_SIZE);
+    }
+
+    /**
+     * Returns the thumbnail of this {@link Metacard}. <br/>
+     * Convenience method for <code>
+     * {@link #getAttribute}({@link Metacard#THUMBNAIL})
+     * </code>
+     *
+     * @return the thumbnail for this {@link Metacard}
+     */
+    @Override
+    public byte[] getThumbnail() {
+        return getPropertyAsBinary(Metacard.THUMBNAIL);
+    }
+
+    /**
+     * Returns the content type of this {@link Metacard}. <br/>
+     * Convenience method for <code>
+     * {@link #getAttribute}({@link Metacard#CONTENT_TYPE})
+     * </code>
+     *
+     * @return the content type for this {@link Metacard}
+     */
+    @Override
+    public String getContentTypeName() {
+        return getPropertyAsString(Metacard.CONTENT_TYPE);
+    }
+
+    /**
+     * Returns the content type version of this {@link Metacard}. <br/>
+     * Convenience method for <code>
+     * {@link #getAttribute}({@link Metacard#CONTENT_TYPE_VERSION})
+     * </code>
+     *
+     * @return the content type version for this {@link Metacard}
+     */
+    @Override
+    public String getContentTypeVersion() {
+        return getPropertyAsString(DynamicMetacard.CONTENT_TYPE_VERSION);
+    }
+
+    /**
+     * Returns the content type namespace of this {@link Metacard}. <br/>
+     * Convenience method for <code>
+     * {@link #getAttribute}({@link Metacard#TARGET_NAMESPACE})
+     * </code>
+     *
+     * @return the content type namespace for this {@link Metacard}
+     */
+    @Override
+    public URI getContentTypeNamespace() {
+        return getPropertyAsURI(Metacard.TARGET_NAMESPACE);
+    }
+
+    /**
+     * Returns the value for the given attribute name as a string
+     *
+     * @param attributeName
+     * @return string value of the given attribute, or null
+     */
+    private String getPropertyAsString(String attributeName) {
+        String s = null;
+        Object o = attributesBean.get(attributeName);
+        if (o instanceof String) {
+            s = (String) o;
+        } else {
+            LOGGER.warn("Unexpected data type retrieving {}: {}",
```

Comment:
All these log statements appear to be the same.  Perhaps use a single log function.